CMakeLists.txt

```diff
@@ -27,14 +27,24 @@ add_definitions(-DAPI_NAME="Vulkan")
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 find_package(PythonInterp 3 REQUIRED)
 
-find_package(VulkanHeaders)
-if(NOT ${VulkanHeaders_FOUND})
-    message(FATAL_ERROR "Could not find Vulkan headers path. This can be fixed by setting VULKAN_HEADERS_INSTALL_DIR to an "
-                        "installation of the Vulkan-Headers repository.")
-endif()
-if(NOT ${VulkanRegistry_FOUND})
-    message(FATAL_ERROR "Could not find Vulkan registry path. This can be fixed by setting VULKAN_HEADERS_INSTALL_DIR to an "
-                        "installation of the Vulkan-Headers repository.")
+if (TARGET Vulkan::Headers)
+    message(STATUS "Using Vulkan headers from Vulkan::Headers target")
+    get_target_property(VulkanRegistry_DIR Vulkan::Registry INTERFACE_INCLUDE_DIRECTORIES)
+else()
+    find_package(VulkanHeaders)
+    if(NOT ${VulkanHeaders_FOUND})
+        message(FATAL_ERROR "Could not find Vulkan headers path. This can be fixed by setting VULKAN_HEADERS_INSTALL_DIR to an "
+                            "installation of the Vulkan-Headers repository.")
+    endif()
+    if(NOT ${VulkanRegistry_FOUND})
```

Comment:
Should this be in its own `if(TARGET Vulkan::Registry)`, outside of `if(TARGET Vulkan::Headers)`?

Reply:
We could do that, but since both Vulkan::Headers and Vulkan::Registry come from the VulkanHeaders package today, that would mean duplicating the find_package() statement as well.

I don't really anticipate a future where these two are separate. The merged build is likely to switch to generating the headers on the fly from the XML next, but even then we'd still have both targets coming from the same project.

I have no strong opinions here, so if you think it's preferable I can move this around.